FridgeExercise/Program.cs

```diff
@@ -1,2 +1,242 @@
+			////refrigerator.PutItemInTheFridge(new Item("chocolate", 2024, 10, 20, 5, "Dairy", "Drink"), refrigerator.Shelves);
+			//refrigerator.PutItemInTheFridge(new Item("milk", 2023, 10, 25, 10, "Dairy", "Drink"), refrigerator.Shelves);
+			////refrigerator.PutItemInTheFridge(new Item("meat", 2023, 10, 27, 15, "Meat", "Drink"), refrigerator.Shelves);
+			//refrigerator.PutItemInTheFridge(new Item("meat2", 2023, 10, 31, 2, "Meat", "Drink"), refrigerator.Shelves);
+			//refrigerator.PutItemInTheFridge(new Item("chily1", 2023, 10, 24, 10, "Parve", "Drink"), refrigerator.Shelves);
+			////refrigerator.PutItemInTheFridge(new Item("milk0", 2023, 10, 20, 19, "Dairy", "Drink"), refrigerator.Shelves);
+			//refrigerator.PutItemInTheFridge(new Item("chily2", 2023, 10, 30, 10, "Parve", "Drink"), refrigerator.Shelves);
+			//refrigerator.PutItemInTheFridge(new Item("meat3", 2023, 11, 01, 1, "Meat", "Drink"), refrigerator.Shelves);
+
+
+			int action =0;
+			string takeOutItemName="",typeToEat="",KosherTypeToEat="";
+			
+
+			while (true)
+			{
+				Console.WriteLine("Press 1: the program will print all the items on the refrigerator and all its contents.");
+				Console.WriteLine("Press 2: the program will print how much space is left in the fridge");
+				Console.WriteLine("Press 3: The program will allow you to put an item in the fridge");
+				Console.WriteLine("Press 4: The program will allow you to take out an item from the fridge");
+				Console.WriteLine("Press 5: the program will clean the refrigerator and print all the checked items .");
+				Console.WriteLine("Press 6: the program will ask you: What do you want to eat?; And the program will bring the product.");
+				Console.WriteLine("Press 7: the program will print all the products sorted by their expiration date.");
+				Console.WriteLine("Press 8: the program will print all the shelves arranged according to the free space left on them.");
+				Console.WriteLine("Press 9: the program will print all the refrigerators arranged according to the free space left in them");
+				Console.WriteLine("Press 10: The program will prepare the refrigerator for shopping");
+				Console.WriteLine("Press 100: system shutdown.");
+
+				action = 0;
+
+				try
+				{
+					action = Int32.Parse(Console.ReadLine());
+				}
+				catch
+				{
+					Console.WriteLine("Please press a number below");
+					continue;
+				}
+				
+				switch (action)
+				{
+					case 1:
+						ThisProgramObject.PrintAllInTheFridge(refrigerator);
+						break;
+					case 2:
+						Console.WriteLine("There are  "+ refrigerator.SpaceIsLeftInTheFridge(refrigerator.Shelves) + " square meters");
+						break;
+					case 3:
+						ThisProgramObject.InsertItem(refrigerator);
+						
+						break;
+					case 4:
+						var temp = 0;
+						Console.WriteLine("What is the item name that you want to take out");
+						takeOutItemName=Console.ReadLine();
+						foreach(Shelf shelf in refrigerator.Shelves)
```

Comment:
```suggestion
						foreach (Shelf shelf in refrigerator.Shelves)
```

Styling :)
And generally, in addition to separating the whole case into a method of its own, I would also move this logic over to the refrigerator. It should have a public method that allows getting the output you're getting here, but shouldn't expose things like the shelves and the items. I wouldn't even let the refrigerator know of the items.
This will basically creating a hierarchy, in which every class is aware of the things within its responsibility, and anything else gets passed on.